storage: properly skip during GC

tbg opened the pull request:
Since FinalScore is almost never zero, this code path would never
be activated. What we want is to skip replicas that do not qualify
for being queued instead.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/16009)
<!-- Reviewable:end -->

tschottdorf commented:
Ah, how awkward. This actually prevents the tests which expect GC to run from carrying it out because the score is too low. Oh well, let me see about that. I think we can do without this in 1.0.1, too.

tbg committed:
storage: properly skip during GC

Since FinalScore is almost never zero, this code path would never
be activated. What we want is to skip replicas that do not qualify
for being queued instead.

tschottdorf commented:
Ok, should be green now(ish).

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

I don't think we need this for 1.0.1 - it only affects ranges whose scores change between shouldQueue and process, right?

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16009#-:-KkRuYvO-uf7fMD13a5A:b7onwtt)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Yes, I'm fine not including it.

tschottdorf commented:
Btw, this does happen often enough that I can find it in lightstep pretty easily. I'll keep my eyes peeled on why that is.
